src/Microsoft.DotNet.Build.Tasks.Feed/BlobFeedAction.cs

```diff
@@ -112,6 +140,71 @@ public BlobFeedAction(string expectedFeedUrl, string accountKey, MSBuild.TaskLog
             return !Log.HasLoggedErrors;
         }
 
+        public async Task UploadAssets(ITaskItem item, SemaphoreSlim clientThrottle, bool allowOverwrite = false)
+        {
+            string relativeBlobPath = item.GetMetadata("RelativeBlobPath");
+            if (string.IsNullOrEmpty(relativeBlobPath))
+            {
+                string fileName = Path.GetFileName(item.ItemSpec);
+                string recursiveDir = item.GetMetadata("RecursiveDir");
+                relativeBlobPath = $"{feed.RelativePath}{recursiveDir}{fileName}";
+            }
+
+            relativeBlobPath = relativeBlobPath.Replace("\\", "/");
+
+            Log.LogMessage($"Uploading {relativeBlobPath}");
+
+            await clientThrottle.WaitAsync();
+
+            bool blobExists = await feed.CheckIfBlobExists(relativeBlobPath);
```

Comment:
We don't need to check if the blob exists if we are allowing overwrite. It just doing more work then we need to.